### ISSUE TITLE
"await" is a reserved word in module context

### ISSUE THREAD

**Issue by jrvidal** [eslint/espree#151]
[11.6.2.2 Future Reserved Words](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-future-reserved-words)

**Comment by nzakas** [comment]
I'm sorry, there's no enough information in this issue to do anything. Are you reporting a bug? If so, what code were you parsing and what was the expected output?

If not, please explain the purpose of this issue.

**Comment by jrvidal** [comment]
Sorry about my cryptic report!

Example:

```js
espree.parse("var await = 1;", {ecmaFeatures: {modules: true}})
```

I would expect this to throw, since `await` is a "future reserved word" in module context. I think I can submit a PR for this, it's a pretty simple fix.

**Comment by nzakas** [comment]
Gotcha! PR definitely welcome.
